lib/router/mbaas.js

```diff
@@ -1,37 +1,71 @@
 'use strict';
 
-var express = require('express')
-  , config = require('../user/config-user')
-  ;
+const express = require('express'),
+  config = require('../user/config-user'),
+  q = require('q');
+
 
 function initRouter(mediator) {
+  /**
+   * @param {String} username Unique identifier of the user.
+   * @param {String} password passed in password to be verified against user.
+   *
+   */
+  var auth = function(username, password) {
+    console.log('Checking credentials for user ' + username);
+    var defered = q.defer();
+    return mediator.request('wfm:user:auth', {username: username, password: password}, {uid: username})
+      .then(function(authenticated) {
```

Comment:
I think this is wrong..

Can you break this up to separate functions that return promises.

I think it should be

```
var auth = function(username, password) {

    function checkAuthentication(authenticated) {
      var checkAuthDeferred = q.defer();
      if (authenticated) {
        console.log(username, ' authenticated');
        return mediator.request('wfm:user:username:read', username)
      }

      console.log('Invalid credentials for ', username);
      checkAuthDeferred.reject('Invalid credentials for ' + username);
      return checkAuthDeferred.promise;
    }

    console.log('Checking credentials for user ' + username);

    return mediator.request('wfm:user:auth', {username: username, password: password}, {uid: username})
      .then(checkAuthentication);
  };
```